analysis_engine/td/extract_df_from_redis.py

```diff
@@ -44,46 +44,193 @@ def extract_option_calls_dataset(
     :param work_dict: dictionary of args
     :param scrub_mode: type of scrubbing handler to run
     """
-    return extract_option_dataset('calls', work_dict, scrub_mode=scrub_mode)
-# end of extract_option_calls_dataset
+    label = f'{work_dict.get("label", "extract")}'
+    ds_id = work_dict.get('ticker')
+    df_type = td_consts.DATAFEED_TD_CALLS
+    df_str = td_consts.get_datafeed_str_td(df_type=df_type)
+    redis_key = work_dict.get(
+        'redis_key',
+        work_dict.get('tdcalls', 'missing-redis-key'))
+    s3_key = work_dict.get(
+        's3_key',
+        work_dict.get('tdcalls', 'missing-s3-key'))
+    redis_host = work_dict.get(
+        'redis_host',
+        None)
+    redis_port = work_dict.get(
+        'redis_port',
+        None)
+    redis_db = work_dict.get(
+        'redis_db',
+        ae_consts.REDIS_DB)
+    verbose = work_dict.get(
+        'verbose_td',
+        False)
 
+    if verbose:
+        log.info(
+            f'{label} - {df_str} - start - redis_key={redis_key} '
+            f's3_key={s3_key}')
 
-def extract_option_puts_dataset(
-        work_dict,
-        scrub_mode='sort-by-date'):
-    """extract_option_puts_dataset
+    if not redis_host and not redis_port:
+        redis_host = ae_consts.REDIS_ADDRESS.split(':')[0]
+        redis_port = ae_consts.REDIS_ADDRESS.split(':')[1]
 
-    Extract the TD options puts for a ticker and
-    return it as a ``pandas.Dataframe``
+    exp_date_str = None
+    calls_df = None
+    status = ae_consts.NOT_RUN
+    try:
+        redis_rec = redis_get.get_data_from_redis_key(
+            label=label,
+            host=redis_host,
+            port=redis_port,
+            db=redis_db,
+            password=work_dict.get('password', None),
+            key=redis_key,
+            decompress_df=True)
 
-    :param work_dict: dictionary of args
-    :param scrub_mode: type of scrubbing handler to run
-    """
-    return extract_option_dataset('puts', work_dict, scrub_mode=scrub_mode)
-# end of extract_option_puts_dataset
+        status = redis_rec['status']
+        if verbose:
+            log.info(
+                f'{label} - {df_str} redis get data key={redis_key} '
+                f'status={ae_consts.get_status(status=status)}')
 
+        if status == ae_consts.SUCCESS:
+            calls_json = None
+            if 'calls' in redis_rec['rec']['data']:
+                calls_json = redis_rec['rec']['data']['calls']
+            else:
+                calls_json = redis_rec['rec']['data']
+            if verbose:
+                log.info(f'{label} - {df_str} redis convert calls to df')
+            exp_date_str = None
+            try:
+                calls_df = pd.read_json(
+                    calls_json,
+                    orient='records')
+                if len(calls_df.index) == 0:
```

Comment:
wouldn't the pandas `read_json` method with `orient='records'` return a list? so the `.index` would be pointing to a function object and taking the len of that would be invalid? http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html